docs/website/docs/user-guides/v3-migration-guide.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Migrate from v2 to v3
+sidebar_position: 2
+---
+
+### Migrate existing odo component from v2 to v3
+If you have created an odo component using odo v2, this section will help you move it to use odo v3.
+1. `cd` into the component directory, and delete the component from the Kubernetes cluster:
+    ```shell
+    odo delete
+   ```
+2. Download and install odo v3 by following the steps mentioned [here](../overview/installation.md).
+3. Use [`odo dev`](../command-reference/dev.md) to start developing your application using odo v3.
+4. You can also use `odo list` to see a list of components that are running on the cluster and which version of odo
+   was used to create them.
+
+If you face any problem, [open an issue on odo's repository](https://github.com/redhat-developer/odo/issues/new?assignees=&labels=&template=Bug.md).
+
+### Where are `odo push` and `odo watch` commands?
+`odo push` and `odo watch` have been replaced in v3 by a single command - `odo dev`. It does the job of two commands 
+in a single command, and also allows the user to not watch if `--no-watch` flag is passed. 
+
+In v2, if you wanted to automatically sync the code on developer system with the application running on a Kubernetes
+cluster, you had to perform two steps - first `odo push` to start the application on the cluster, and second `odo watch`
+to automatically sync the code. In v3, `odo dev` performs both actions with a single command.
```

Comment:
Too many sentences that can be summed up as:

`odo dev` now performs both `odo push` and `odo watch` actions as one command.

Reply:
I prefer not to assume the user to be too smart. They can skim through the part or ignore it if they already understand. But for general purpose, I think this would be useful.